daliuge-engine/dlg/remote.py

```diff
@@ -103,10 +103,7 @@ def createClient(host, username=None, pkeyPath=None):
     client = SSHClient()
     client.set_missing_host_key_policy(AutoAddPolicy())
 
-    pkey = (
-        RSAKey.from_private_key_file(os.path.expanduser(pkeyPath)) if pkeyPath else None
-    )
-    client.connect(host, username=username, pkey=pkey)
+    client.connect(host, username=username, key_filename=pkeyPath)
```

Comment:
I assume an `""` empty string for `key_filename` will be treated the same as `None`? That latter is the default in their API, while the former is the default for the new `--ssh_key` option. I assume you've tested this and it's working as expected anyway.